tests/mtest/src/main/scala/tests/BaseSuite.scala

```diff
@@ -32,18 +40,41 @@ class BaseSuite extends munit.FunSuite with Assertions {
 
   override def munitTimeout: Duration = Duration("10min")
 
-  // NOTE(olafur): always ignore flak test failures.
+  // NOTE(olafur): always ignore flaky test failures.
   override def munitFlakyOK = true
-  override def munitRunTest(
-      options: TestOptions,
-      body: => Any
-  ): Any = {
-    if (Properties.isWin && options.tags(FlakyWindows)) {
-      munitFlaky(options, body)
-    } else {
-      super.munitRunTest(options, body)
-    }
-  }
+
+  case class Rerun(count: Int) extends munit.Tag("Rerun")
+
+  override def munitTestTransforms: List[TestTransform] =
+    super.munitTestTransforms ++ List(
+      new TestTransform("Rerun", { test =>
+        val rerunCount = test.tags
+          .collectFirst { case Rerun(n) => n }
+          .getOrElse(1)
+        if (rerunCount == 1) test
+        else {
+          test.withBody(() => {
+            Future.sequence(1.to(rerunCount).map(_ => test.body()).toList)
+          })
+        }
+      }),
+      new TestTransform("Flaky Windows", { test =>
+        if (Properties.isWin && test.tags.contains(FlakyWindows)) {
+          test.withBodyMap(_.transformCompat {
+            case Success(value) => Success(value)
+            case Failure(exception) =>
+              if (munitFlakyOK) {
+                Success(new TestValues.FlakyFailure(exception))
+              } else {
+                throw exception
+              }
+          }(munitExecutionContext))
```

Comment:
@olafurpg is there a way to reuse the default implementation (like we did in 0.5 using `munitFlaky`?

Reply:
`munitFlakyTransform` is the default transform, which you can apply as a `Test => Test` function

Reply:
An alternative approach is to add the `Flaky` tag based on `Properties.isWin`

```scala
new TestTransform("FlakyWindows", test => if (test.tags(FlakyWindows) && Properties.isWin) test.tag(Flaky) else test)
```
As long as that transform comes before `munitFlakyTransform` then you don't need to re-implement the `.flaky` logic

Reply:
Ah, cool. I'll do that.